src/NuGet.Core/NuGet.Versioning/VersionRangeFactory.cs

```diff
@@ -233,6 +236,32 @@ public static bool TryParse(string value, bool allowFloating, out VersionRange v
                 }
             }
 
+            // Illogical version range detection
+            if (minVersion != null && maxVersion != null
+                && minVersion >= maxVersion
+                && !(minVersion == maxVersion && isMinInclusive && isMaxInclusive))  // Exclude (9.0.0,9.0.0) since it's used as empty version range.
+            {
+                if (partsLength == 1)
+                {
+                    var onePartValidation = isMinInclusive ^ isMaxInclusive; // (1.0.0] and [1.0.0) are invalid
```

Comment:
As mentioned above, you can simplify this by writing it as

> !(isMinInclusive && isMaxInclusive)

You can check this with a truth table or a quick console app. 

```cs
static void Main(string[] args)
        {
            Compare(true, true);
            Compare(true, false);
            Compare(false, true);
            Compare(false, false);

        }

        public static void Compare(bool isMinInclusive, bool isMaxInclusive)
        {
            if (Calculate(isMinInclusive, isMaxInclusive) != CalculateSimple(isMinInclusive, isMaxInclusive))
            {
                Console.WriteLine(isMinInclusive + " " + isMaxInclusive + " Original: " + Calculate(isMinInclusive, isMaxInclusive) + " Updated: " + CalculateSimple(isMinInclusive, isMaxInclusive));

            }
        }

        private static bool CalculateSimple(bool isMinInclusive, bool isMaxInclusive)
        {
            return !(isMinInclusive && isMaxInclusive);
        }

        public static bool Calculate(bool isMinInclusive, bool isMaxInclusive)
        {
            var onePartValidation = isMinInclusive ^ isMaxInclusive;
            onePartValidation |= !isMinInclusive && !isMaxInclusive;
            return onePartValidation;

        }
```

Reply:
Taken your suggestion.